internal/app/machined/pkg/system/services/extension_test.go

```diff
@@ -0,0 +1,35 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package services_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/talos-systems/talos/internal/app/machined/pkg/system/services"
+	extservices "github.com/talos-systems/talos/pkg/machinery/extensions/services"
+)
+
+// TestGetOCIOptions ensures that no oci option spec is created for the options MaskedPaths and ReadonlyPaths if they are set to nil. This is backward compatible behavior.
+func TestGetOCIOptions(t *testing.T) {
+	// given
+	svc := &services.Extension{
+		Spec: &extservices.Spec{
+			Container: extservices.Container{
+				Security: extservices.Security{
+					MaskedPaths:   nil,
```

Comment:
I don't think we want masked paths to be nil by default, since containerd sets up default maskedpaths, so i think we need  it the other way around, @smira did I understand this this right?

Reply:
I tried to emulate the case where the user didn't specify any maskedPaths in the yaml spec. In that case MaskedPaths here will be nil and as a result we want to fall back to the default MaskedPaths of containerd like you said. That's why I want to test, that the MaskedPaths option doesn't get added to the list.

Reply:
so how i do set empty masked paths like https://github.com/siderolabs/talos/pull/5402/files#diff-58056f2700d6a6e74200d7eaf5e8e2466f0eb2f368a4a1d540efb1a357fa6c1fR145, becuase by default talos won't set them and just using

```yaml
security:
  maskedPaths: []
```

feels like it's confusing. Maybe `withEmptyMaskedPaths`, I'm not sure which is better

Reply:
I think specifying empty `maskedPaths: []` is fine, we just need to document that default value is non-empty and OCI default